src/main/java/games/strategy/triplea/ui/PlacePanel.java

```diff
@@ -88,9 +88,7 @@ public void territorySelected(final Territory territory, final MouseDetails e) {
       final UnitChooser chooser = new UnitChooser(units, Collections.emptyMap(), getData(),
           false, getMap().getUIContext());
       final String messageText = "Place units in " + territory.getName();
-      if (maxUnits[0] > 0) {
-        chooser.setMaxAndShowMaxButton(maxUnits[0]);
-      }
+      chooser.setMaxAndShowMaxButton(maxUnits[0]);
```

Comment:
If you watch the video I linked on #646, you will see, that if this method is not executed when maxunits = 0 if we open the frame, we can choose unlimited units just to see an error dialog in our face because that's an invalid thing to do...